app/src/main/java/com/swent/assos/model/service/impl/DbServiceImpl.kt

```diff
@@ -1,13 +1,26 @@
 package com.swent.assos.model.service.impl
 
 import com.google.firebase.firestore.FirebaseFirestore
+import com.swent.assos.model.data.Association
 import com.swent.assos.model.service.AuthService
 import com.swent.assos.model.service.DbService
 import javax.inject.Inject
+import kotlinx.coroutines.tasks.await
 
 class DbServiceImpl
 @Inject
 constructor(
     private val firestore: FirebaseFirestore,
     private val auth: AuthService,
-) : DbService {}
+) : DbService {
+
+  override suspend fun getAllAssociations(): List<Association> {
+    return firestore.collection("associations").get().await().documents.map { doc ->
+      val fields = doc.data!!
```

Comment:
important: the returned doc could be null so it would be a good idea to check it, and if so to report it via [logs](https://developer.android.com/reference/android/util/Log) and/or maybe show a [toast message](https://developer.android.com/guide/topics/ui/notifiers/toasts) to the user stating that it failed.